File: relatorio/codigos/ruby/micro10.rb
# Calcula o fatorial de um numero

def fatorial(n)
  if n <= 0
    return 1
  else
    return (n * fatorial(n-1))
  end
end

puts("Digite um numero:")
numero = gets.chomp.to_i
fat = fatorial(numero)

puts("O fatorial de #{numero} e: #{fat}")

File: relatorio/codigos/lua/micro10.lua
--Listagem 22: Calcula o fatorial de um numero

--[[ Funcao: recebe um numero e calcula recursivamente o fatorial desse número --]]

function fatorial(n)
	 if(n <= 0)
	 then return 1
	 else return (n* fatorial(n-1))
	 end
end

print("Digite um numero: ")
numero = tonumber(io.read())
fat = fatorial(numero)

print("O fatorial de", numero, "e: ", fat)